#1909 removed active effect for xp adjustments

Greenboat24 opened the pull request:
Fixes #1909 at the cost of making it not possible to undo an XP adjustment, but the fix for that is just to re-adjust.

wrycu commented:
two small things:
1. can you please update the readme with your change?
2. I don't recall for sure off the top of my head, but `update` is _generally_ an async operation. I don't see an await. are you sure it's synchronous?

overall, LGTM. though your wall of text on Discord indicates we may need further changes.

Greenboat24 commented:
Okay, I've added the await, though XP Grants doesn't await the update either. 

I've also updated the changelog to include the fix. I assume that's what you meant by adding to the readme? 

Also yeahhhh sorry about the wall of text, I figured it would be best to give the full context of what I found rather than leave it ambiguous. I'll leave my walls of text to github comments from now on instead of polluting the discord :) 

In terms of further changes, I think it might be best if the XP ActiveEffects were removed completely and replaced by something like a purchases array, and then turn availableXP into a derived value. ActiveEffects in general might need looking at though because there appears to be other problems around characteristic purchases not applying with other derived values like soak, whereas general updates do. Happy to discuss!
